### ISSUE TITLE
bug: module.lacework_cfg_iam_role should pin a specific version

### ISSUE THREAD

**Issue by jrobison-sb** [lacework/terraform-aws-config#86]
**Describe the bug**
This module consumes the `lacework/iam-role/aws` module, but [doesn't pin a specific version of that module](https://github.com/lacework/terraform-aws-config/blob/main/main.tf#L16). This means that whenever the `iam-role` module ships a new version, users of the  `terraform-aws-config` module will inherit that new version of `iam-role` without really knowing it.

**Steps to reproduce**

Take any terraform environment where you have used `terraform-aws-config` and you have already run `terraform init` before https://github.com/lacework/terraform-aws-iam-role/pull/46 was merged, run `terraform plan` and see no diffs.

Then run `rm -rf .terraform && terraform init && terraform plan` which will pull in the new version of `iam-role` and you'll see diffs.

**Expected behavior**

You should never see diffs without explicitly bumping a version.

**Screenshots**
N/A

**Please complete the following information):**
 - Terraform Version: v1.6.1
 - Module Version: 0.13.0

**Comment by mgordziejewksi-clari** [comment]
Please fix it, it just broke my modules.

**Comment by mgordziejewksi-clari** [comment]
One thing is upgrading the provider in one of the versions of your module, another is keeping the versions pinned in versions of your module. Right now, if you use old version of your module, it is entirely broken if you depend on 1.0 version in any place. I think it's just good practice that old versions of your module don't break out of sudden.

**Comment by PengyuanZhao** [comment]
@mgordziejewksi-clari Definitely. 1.0 stopped working due to a breaking change in one of its dependencies. We had to force all customers to use 2.0 Sorry for the trouble. Here is a related thread https://github.com/lacework/terraform-aws-iam-role/issues/60
